lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -314,15 +314,26 @@ LinearLayout sharedToLinearLayoutLeadingOffset(ArrayRef<int64_t> shape,
       bases2D.push_back({row, vec * ((row / perPhase) % maxPhase)});
     }
   }
+
+  // Replicate the tiles to fill out the inner dims of shapePerCTA.
+  for (int logRow = llvm::Log2_32(tileRows);
+       logRow < llvm::Log2_32(shapePerCTA[rowDim]); logRow++) {
+    bases2D.push_back({1 << logRow, 0});
+  }
+
   LinearLayout tileLayout =
       LinearLayout({{S("offset"), bases2D}}, {rowDimName, colDimName});
-
   // Add the remaining dimensions.
   for (int dim = batchDims - 1; dim >= 0; --dim) {
     tileLayout *= LinearLayout::identity1D(shapePerCTA[dim], S("offset"),
                                            outDimNames[dim]);
   }
 
+  for (int logCol = llvm::Log2_32(tileCols);
+       logCol < llvm::Log2_32(shapePerCTA[colDim]); logCol++) {
+    bases2D.push_back({0, 1 << logCol});
+  }
+
```

Comment:
this is not even used?

Reply:
ah yeah this one is useless and done by `combineCtaCgaWithShape` although I wanted to pick the order explicitly, not sure if there is a way to do that. Since `combineCtaCgaWithShape` will put this one at the end it is fine

Reply:
my point was that at this point you are not using `bases2D` anymore as you have already created the layout. I think `CtaCgaWithShape` was doing this for you (order is not importatn when there is just one dimension to reshape) and that's why you were seeing the expected behaviour.